### PR TITLE
add requests to API && translate

### PR DESCRIPTION
**Pull request by pakoval**

### CODE REVIEW

File: src/translation/request.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  "https://translate.googleapis.com/translate_a/single?client=gtx&sl=";
+
+export async function createURL(source: string, target: string, text: string) {
+  const url = `${
+    BASE_URL + source + "&tl=" + target + "&dt=t&q=" + encodeURI(text)
+  }`;
+
+  try {
+    const { data } = await axios.get(url);
+    // data[0][0][0] here gets results of translation
```
Comment (silchencko) [inline src/translation/request.ts]: // data[0] - full translation config/details/info
// data[0][0] - translation result
// data[0][0] - translation in the target language
something like this
or 'the first [0] means ..., the second [0] means...

File: src/translation/request.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return data[0][0][0];
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export async function getPhones() {
```
Comment (silchencko) [inline src/translation/request.ts:26]: is it for testing purpose only? Or do you have plans to use it?

File: src/views/HomeView.vue [hunk 3]
```diff
@@ -18,28 +18,32 @@
...
-      this.post = data;
-    } catch (error) {
-      console.log(error);
-    }
+
+  sendTextDebounce = debounce(this.sendText, 1000);
+  async sendText() {
+    this.outputTextarea = await createURL("en", "uk", this.inputTextarea);
+  }
+  mounted() {
```
Comment (silchencko) [inline src/views/HomeView.vue]: async await

File: src/translation/request.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+const BASE_URL =
+  "https://translate.googleapis.com/translate_a/single?client=gtx&sl=";
+
+export async function createURL(source: string, target: string, text: string) {
```
Comment (silchencko) [inline src/translation/request.ts]: does this function create URL? I expected it would translate...

File: src/views/HomeView.vue [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
-        <TextArea v-model="inputTextarea" name="inputTextarea" />
+        <TextArea v-model="inputTextarea" @input="sendTextDebounce" />
         <p class="textarea-block__length">
           {{ inputTextarea.length }} / {{ maxLength }}
         </p>
       </div>
 
       <div class="textarea-block">
-        <TextArea />
+        <TextArea :value="outputTextarea" />
```
Comment (silchencko) [inline src/views/HomeView.vue:12]: maybe has sense to make it readonly

File: src/translation/request.ts (hunk 1 above)
Comment (pakoval) [reply src/translation/request.ts]: okay, I got it

File: src/translation/request.ts (hunk 2 above)
Comment (pakoval) [reply src/translation/request.ts:26]: 1. For testing purpose
2. I think to create another page for render data from dummyAPI

File: src/views/HomeView.vue (hunk 3 above)
Comment (pakoval) [reply src/views/HomeView.vue]: like this?
async mounted() {
    await getPhones();
    await addProduct();
  }

File: src/translation/request.ts (hunk 4 above)
Comment (pakoval) [reply src/translation/request.ts]: renamed

File: src/views/HomeView.vue (hunk 5 above)
Comment (pakoval) [reply src/views/HomeView.vue:12]: good idea

File: src/views/HomeView.vue (hunk 3 above)
Comment (silchencko) [reply src/views/HomeView.vue]: yes, you could combine all async functions in `Promise.all` but I'm almost sure, you will move some methods from `mounted` to a different places in the code

**Review by silchencko — APPROVED** [review]

**Pull request merged**